alertwildfire_live_recording.py

```diff
@@ -127,40 +129,98 @@ def fetchAllCameras(camera_names_to_watch):
+    #    for folder in goog_helper.driveListFilesByName(googleServices['drive'], settings.alertwildfire_archive):
+    #        for fileobj in goog_helper.driveListFilesByName(googleServices['drive'], folder['id']):
+    #            if fileobj['name'][-23:-4]< current_target:
+    #                logging.error('deleting file', fileobj['name'])
+    #                goog_helper.deleteItem(googleServices['drive'], file_id)
+    #        
+    return True
+
 def main():
     """directs the funtionality of the process ie start a cleanup, record all cameras on 2min refresh, record a subset of cameras, manage multiprocessed recording of cameras
     Args:
+        -c  cleaning_threshold" (flt): time in hours to store data
         -o  cameras_overide    (str): list of specific cameras to watch
         -p  parallelize       (bool): toggle to parallelize
+        -a  agents            (int): number of agents to assign for parallelization
     Returns:
         None
     """
     reqArgs = []
     optArgs = [
+        ["c", "cleaning_threshold", "time in hours to store data"],
         ["o", "cameras_overide", "specific cameras to watch"],
-        ["p", "parallelize", "toggle parallelisation"]
+        ["p", "parallelize", "toggle parallelisation"],
+        ["a", "agents", "number of agents to assign for parallelization"]
     ]
     args = collect_args.collectArgs(reqArgs,  optionalArgs=optArgs, parentParsers=[goog_helper.getParentParser()])
     
+    
+    if args.cleaning_threshold:
+        googleServices = goog_helper.getGoogleServices(settings, args)
+        cleaning_threshold = float(args.cleaning_threshold)
+        cleanup_archive(googleServices, cleaning_threshold)
     if args.cameras_overide:
         listofRotatingCameras = list(args.cameras_overide.replace(" ", "").strip('[]').split(','))
     else:
         listofCameras = alertwildfire_API.get_all_camera_info()
         listofRotatingCameras = [camera["name"] for camera in listofCameras if (camera["name"][-1]=='2') ]
+    if args.agents:
+        agents = int(args.agents)
+    else:
+        agents = None
     if args.parallelize:
-        parallel = args.parallelize
+        parallel = True
+        #num of camera's per process
+        test = "Axis-Briar2"
+        googleServices = goog_helper.getGoogleServices(settings, args)
+        dbManager = db_manager.DbManager(sqliteFile=settings.db_file,
+                                        psqlHost=settings.psqlHost,                  psqlDb=settings.psqlDb,    
+                                        psqlUser=settings.psqlUser, psqlPasswd=settings.psqlPasswd)
+        temporaryDir = tempfile.TemporaryDirectory()
+        tic = time.time()
+        capture_and_record(googleServices, dbManager, temporaryDir.name, test)
+        toc =time.time()-tic
+        # target estimate of camera refresh time
+        target_refresh_time_per_camera = 12#secs
+        num_cameras_per_process = math.floor(target_refresh_time_per_camera / toc)
```

Comment:
This type of timing based work splitting should be based on average timing over multiple requests, and also periodically repeating the measurement and adapting.  That complexity is not worth it for version 1, but something to keep in mind.